### ISSUE TITLE
Import .md (and not .mdx file) as Docs Only page?

### ISSUE THREAD

**Issue by smurrayatwork** [storybookjs/storybook#11981]
If you are reporting a bug or requesting support, start here:
### Bug or support request summary

Is it possible to directly import a normal `.md` file (such as a `README.md` file) into a story for display? Or, is it possible to import a `.md` file as a Docs Only page? If so, how would I go about doing that?

So far, I've only been able to get that working by copying the contents of a `.md file into a `.stories.mdx` file, while also using `<Meta />` for the docs only approach.

### Please specify which version of Storybook and optionally any affected addons that you're running

- @storybook/react 6.0.4
- @storybook/addon-docs 6.0.4

**Comment by shilman** [comment]
Yes, it's possible but not well documented yet. See this PR for details: https://github.com/storybookjs/storybook/pull/11334

**Comment by smurrayatwork** [comment]
@shilman ,

I was able to get it working for .mdx files by setting `transcludeMarkdown: true` for the docs addon's options. Thanks!

Just posting for others' reference:
```
// markdown.stories.mdx
import { Meta } from '@storybook/addon-docs/blocks';
import Readme from './README.md';

<Meta title="about/Markdown Test" />

<Readme />
```

I wasn't able to import the .md file in a `.stories.js` file and maintain the same styling as the `.mdx` file though.

**Comment by smurrayatwork** [comment]
@shilman Is there a way to disable the `Canvas` tab when using transclude?

**Comment by shilman** [comment]
I think you can run `yarn storybook --docs` to disable canvas

**Comment by psychobolt** [comment]
> I think you can run `yarn storybook --docs` to disable canvas

Is there a way to disable canvas for a MDX/MD story or any specific story?

**Comment by smurrayatwork** [comment]
@psychobolt I discovered this myself from a storybook example
```
<Meta
  title="my meta title here" 
  parameters={{ previewTabs: { canvas: { hidden: true } } }} 
/>
```

**Comment by smurrayatwork** [comment]
@shilman ,

With `transcludeMarkdown: true`, is there a way to then pass the imported `.md` file to a DocsPage component somehow? I have a lot of stories that have external `.md` files, and I'd like to be able to pass the component created by having transclude on to either a custom DocsPage component or the existing one. The only way I've been able to get that part working is by using `raw-loader` inline:
``` 
import readme from '!raw-loader!./README.md';

export default {
  parameters: {
    docs: {
      description: { component: readme }
    }
  }
}
```

**Comment by shilman** [comment]
Hmm not really. You could do the `raw-loader` hack that you're doing.

You could also do a custom DocsPage: https://storybook.js.org/docs/react/writing-docs/docs-page#remixing-docspage-using-doc-blocks

```
import Readme from './README.md'

export default {
  parameters: {
    docs: {
      page: () => (
        <>
          <Title />
          <Subtitle />
          <Readme />
          <Primary />
          <ArgsTable story={PRIMARY_STORY} />
          <Stories />
       </>
      )
    }
  }
};
```

Not very satisfying I know...

**Comment by psychobolt** [comment]
```md
import { Meta, Description } from '@storybook/addon-docs/blocks';

import Readme from '!raw-loader!../../README.md';

<Meta title="Documentation/Readme" />

<Description markdown={Readme} />
```

Seems like the raw-loader also works on my case.

**Comment by peterpeterparker** [comment]
Import  .md files  as displayed above, with the help of mdx  or raw loader, works fine, thank you all for your answers!
But, the block of code (documented in the .md files) are not highlighted once rendered (at least in my case).

Anyone face(d) the same issue and managed to solve it?

Example: https://docs.deckdeckgo.com/?path=/docs/templates-title--title

Source: https://github.com/deckgo/deckdeckgo/blob/main/docs/src/stories/templates/Title.stories.js

Mdx: https://github.com/deckgo/deckdeckgo/blob/main/docs/src/stories/templates/Readme.mdx

Md: https://github.com/deckgo/deckdeckgo/blob/main/webcomponents/slides/title/src/components/slide/readme.md

**Comment by csantos1113** [comment]
> Hmm not really. You could do the `raw-loader` hack that you're doing.
> 
> You could also do a custom DocsPage: https://storybook.js.org/docs/react/writing-docs/docs-page#remixing-docspage-using-doc-blocks
> 
> ```
> import Readme from './README.md'
> 
> export default {
>   parameters: {
>     docs: {
>       page: () => (
>         <>
>           <Title />
>           <Subtitle />
>           <Readme />
>           <Primary />
>           <ArgsTable story={PRIMARY_STORY} />
>           <Stories />
>        </>
>       )
>     }
>   }
> };
> ```
> 
> Not very satisfying I know...

Hi, I have this version working, but I'm having issues with my IDEs (IntelliJ and VSCode) about the TS definitions:
```
TS2307: Cannot find module '../README.md' or its corresponding type declarations.
```

**Comment by TimJJTing** [comment]
> Import .md files as displayed above, with the help of mdx or raw loader, works fine, thank you all for your answers! But, the block of code (documented in the .md files) are not highlighted once rendered (at least in my case).
> 
> Anyone face(d) the same issue and managed to solve it?
> 
> Example: https://docs.deckdeckgo.com/?path=/docs/templates-title--title
> 
> Story: https://github.com/deckgo/deckdeckgo/blob/main/docs/src/stories/templates/Title.stories.js
> 
> Mdx: https://github.com/deckgo/deckdeckgo/blob/main/docs/src/stories/templates/Readme.mdx
> 
> Md: https://github.com/deckgo/deckdeckgo/blob/main/webcomponents/slides/title/src/components/slide/readme.md

I'm also facing the same issue with `.mdx` file.
In the end I gave up and use another approach which might looks funny:

```jsx
// Introduction.stories.js
import React from "react";
import { Stories } from "@storybook/addon-docs";
import ReactMarkdown from "react-markdown";
import remarkGfm from "remark-gfm";
import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
import { vscDarkPlus } from "react-syntax-highlighter/dist/esm/styles/prism";
import Readme from "../../README.md";

export default {
  title: "Hooks/Introduction",
  parameters: {
    docs: {
      page: () => <Stories />,
    },
  },
};

export const Introduction = () => {
  return (
    <ReactMarkdown
      remarkPlugins={[[remarkGfm, { singleTilde: false }]]}
      children={Readme}
      components={{
        code({node, inline, className, children, ...props}) {
          const match = /language-(\w+)/.exec(className || '')
          return !inline && match ? (
            <SyntaxHighlighter
              children={String(children).replace(/\n$/, '')}
              style={vscDarkPlus}
              language={match[1]}
              PreTag="div"
              {...props}
            />
          ) : (
            <code className={className} {...props}>
              {children}
            </code>
          )
        }
      }}
    />
  );
};

```

**Comment by mathiasgheno** [comment]
> Import .md files as displayed above, with the help of mdx or raw loader, works fine, thank you all for your answers! But, the block of code (documented in the .md files) are not highlighted once rendered (at least in my case).
> 
> Anyone face(d) the same issue and managed to solve it?
> 
> Example: https://docs.deckdeckgo.com/?path=/docs/templates-title--title
> 
> Story: https://github.com/deckgo/deckdeckgo/blob/main/docs/src/stories/templates/Title.stories.js
> 
> Mdx: https://github.com/deckgo/deckdeckgo/blob/main/docs/src/stories/templates/Readme.mdx
> 
> Md: https://github.com/deckgo/deckdeckgo/blob/main/webcomponents/slides/title/src/components/slide/readme.md

I had this problem in my project too. I solved this problem installing `@storybook/addon-docs` in the root of the monorepo and I also deleted the `@storybook/addon-docs` folder  in my package trough `prepare` script in my storybook package. I think the reason is the configuration of `main.js` that is linking `.mdx` files [outside the root of the storybook package](https://github.com/deckgo/deckdeckgo/blob/main/docs/.storybook/main.js#L2).

**Comment by determilove** [comment]
```mdx
import { Markdown } from "@storybook/addon-docs/blocks";
import { Meta } from "@storybook/blocks";
import XXXXX from "./xxxxx.md?raw";

<Meta title="xxxxx" />

<Markdown>{XXXXX}</Markdown>
```

It works for me (Storybook v8.6.14)

Note the **"?raw"**

[https://storybook.js.org/docs/api/doc-blocks/doc-block-markdown](https://storybook.js.org/docs/api/doc-blocks/doc-block-markdown)
